asv_bench/benchmarks/eval.py

```diff
@@ -7,64 +7,61 @@
 
 
 class Eval(object):
+
     goal_time = 0.2
 
     params = [['numexpr', 'python'], [1, 'all']]
     param_names = ['engine', 'threads']
 
     def setup(self, engine, threads):
-        self.df = DataFrame(np.random.randn(20000, 100))
-        self.df2 = DataFrame(np.random.randn(20000, 100))
-        self.df3 = DataFrame(np.random.randn(20000, 100))
-        self.df4 = DataFrame(np.random.randn(20000, 100))
+        np.random.seed(1234)
+        self.df = pd.DataFrame(np.random.randn(20000, 100))
+        self.df2 = pd.DataFrame(np.random.randn(20000, 100))
+        self.df3 = pd.DataFrame(np.random.randn(20000, 100))
+        self.df4 = pd.DataFrame(np.random.randn(20000, 100))
 
         if threads == 1:
             expr.set_numexpr_threads(1)
 
     def time_add(self, engine, threads):
-        df, df2, df3, df4 = self.df, self.df2, self.df3, self.df4
-        pd.eval('df + df2 + df3 + df4', engine=engine)
+        pd.eval('self.df + self.df2 + self.df3 + self.df4', engine=engine)
 
     def time_and(self, engine, threads):
-        df, df2, df3, df4 = self.df, self.df2, self.df3, self.df4
-        pd.eval('(df > 0) & (df2 > 0) & (df3 > 0) & (df4 > 0)', engine=engine)
+        pd.eval('(self.df > 0) & (self.df2 > 0) & '
+                '(self.df3 > 0) & (self.df4 > 0)', engine=engine)
 
     def time_chained_cmp(self, engine, threads):
-        df, df2, df3, df4 = self.df, self.df2, self.df3, self.df4
-        pd.eval('df < df2 < df3 < df4', engine=engine)
+        pd.eval('self.df < self.df2 < self.df3 < self.df4', engine=engine)
 
     def time_mult(self, engine, threads):
-        df, df2, df3, df4 = self.df, self.df2, self.df3, self.df4
-        pd.eval('df * df2 * df3 * df4', engine=engine)
+        pd.eval('self.df * self.df2 * self.df3 * self.df4', engine=engine)
 
     def teardown(self, engine, threads):
         expr.set_numexpr_threads()
 
 
 class Query(object):
+
     goal_time = 0.2
 
     def setup(self):
-        self.N = 1000000
-        self.halfway = ((self.N // 2) - 1)
-        self.index = date_range('20010101', periods=self.N, freq='T')
-        self.s = Series(self.index)
+        np.random.seed(1234)
+        self.N = 10**6
```

Comment:
When you are cleaning up, you can also remove such unneeded   `self.`'s, for things that are only used in init (they are a left over from the autogeneration of this code when the benchmarks were translated to asv)